.github/workflows/automate-review-labels.yml

```diff
@@ -82,19 +90,19 @@
               // with different permissions. If it is combined with the following step,
               // then the labeling actions are marked as being completed by whomever
               // created the token.
-              const membership_response = github.rest.teams.getMembershipForUserInOrg({
-                  org: 'conda-forge',
-                  team_slug: 'staged-recipes',
-                  username: '${{ github.event.sender.login }}'
-              });
-              membership_response.then(function(result) {
-                 console.log(result);
-                 if (result.data.state == 'active'){
-                   return;
-                 } else {
-                   throw new Error('Reviewer not active member of conda-forge/staged-recipes');
-                 }
-              });
+              try {
+                github.rest.teams.getMembershipForUserInOrg({
+                    org: 'conda-forge',
+                    team_slug: 'staged-recipes',
+                    username: '${{ github.event.sender.login }}'
+                });
+              } catch (e) {
+                if (e.name == 'HttpError') {
+                 // Reviewer is not a member
+                } else {
+                  throw e; // let other errors bubble up
+                }
+              }
```

Comment:
In this section, I think we should avoid try-catch. It is possible that the response from the server is "this person is not a member" without raising an error.

Furthermore, the response is asynchronous, so I think we need to use continue to use `.then()` in order to wait for the response from the server.

If we want to avoid masking uncaught errors with the custom `Error('Reviewer not an active...')`, then we should raise the Error on the IF instead of the ELSE like this:


```javascript
console.log(result);
if (result.status != 200 or result.data.state != 'active'){
   throw new Error('Reviewer not a member');
}
```

Reply:
From what I can see the response will always be `404` so a full fault if the commenter is not part of the team: https://docs.github.com/en/rest/reference/teams#get-team-membership-for-a-user 
Not sure why GitHub chose not to return a normal response but anyways...

We can try to catch other cases as well but from what I can see there aren't any.

The problem with throwing the error is that this pollutes the notifications of anyone watching the repo as failed runs are added there, unlike what Azure is doing. So we need to finish the job successfully, even if someone is not a member. 
Maybe we can return a variable that will either be true or false and then instead of checking for success check for that variable? Something like in here: https://github.com/actions/github-script#reading-step-results

If you agree, I'll update it accordingly.

Reply:
> pollutes the notifications of anyone watching the repo [...] we need to finish the job successfully, even if someone is not a member.

That's a great point. I didn't realize, it did that.

> Maybe we can return a variable that will either be true or false

Yes. Try this approach. The only reason I didn't try this originally was that this logic cannot go in `if:` I has to go in `script:`? But not having notification spam is better than executing one less step in a job.

Reply:
Ok, will update accordingly. My JS is pretty poor so does your snippet above work already? Then I'm just going to copy/paste and add the Actions logic.

By the way, feel free to push to my branch as well if you want to change something.

Reply:
> My JS is pretty poor so does your snippet above work already?

Almost? I think javascript uses `||`, `&&` instead of `or`, `and`.

Reply:
From what I can see the error is not directly returned so I will try to trigger that in the test repo to find out what the structure of the error message is.